libraries/AP_HAL/UARTDriver.h

```diff
@@ -5,6 +5,11 @@
 #include "AP_HAL_Namespace.h"
 #include "utility/BetterStream.h"
 
+#ifndef HAL_UART_STATS_ENABLED
+#define HAL_UART_STATS_ENABLED !defined(HAL_NO_UARTDRIVER)
```

Comment:
wouldn't it make sense to enable this for !AP_Periph? Solves the periph problem you're trying to fix. Periph's don't need this stuff by default.

Reply:
Good question!

You're probably right - I was just trying to give us a bit of room on the specific device.

If the content can only be retrieved via mavlink FTP we could probably gate on that instead - some periphs *do* support mavlink :-)